nym-vpn-core/crates/nym-gateway-directory/src/gateway_client.rs

```diff
@@ -202,9 +202,10 @@ impl GatewayClient {
 
     pub async fn lookup_all_gateways_from_nym_api(&self) -> Result<GatewayList> {
         let mut gateways = self
-            .lookup_described_gateways()
+            .lookup_described_nodes()
             .await?
             .into_iter()
+            .filter(|node| node.description.declared_role.entry)
```

Comment:
Exit no?

Reply:
you tell me what's the expected behaviour. the old endpoints returned all gateways, i.e. both entry and exit, so I tried to replicate the behaviour

Reply:
Yes, that's what I was thinking, why isn't it entry | exit, instead of just entry?

Reply:
because this is not the role that has been assigned to the node in the contract (this would be the next step), but rather what this node declares it can operate as. and currently exit implies entry, i.e. you must declare you can operate as an entry if you declare yourself exit-capable